Add nix-auto-follow as an allfollow alternative

ginkogruen opened the pull request:
This adds a module which enables [nix-auto-follow](https://github.com/fzakaria/nix-auto-follow) to be used as an alternative to [allfollow](https://github.com/spikespaz/allfollow).

It is currently working but for now it's just a straight copy of the allfollow module with minimum needed adjustments to make it work.

I do get an error if I try to run `write-flake` a second time which I don't understand:
```
❯ nix run .#write-flake
warning: Git tree '/home/ginkogruen/.nixfiles' is dirty
error:
       … while evaluating an expression to select 'drvPath' on it
         at «internal»:1:552:
       … while evaluating strict
         at «internal»:1:552:
       (stack trace truncated; use '--show-trace' to show the full trace)

       error: python3.13-types-Pygments-2.17.0.20240310 does not configure a `format`. To build with setuptools as before, set `pyproject = true` and `build-system = [ setuptools ]`.`
```

I also still need to update the documentation to reflect this change.

Also I'm unsure if there needs to be some kind of mechanism that prevents someone from importing multiple flake formatting modules.

Let me know if there are any other things that need to be considered before the PR can be merged.

vic commented:
Looks good, could you rebase and fix conflicts ? 

Looks like it works the first time because it uses nixpkgs stable, but after it changes its own nixpkgs follow to the root nixpkgs (unstabe) it no longer works.

Can you add the error you get to this issue?
https://github.com/fzakaria/nix-auto-follow/issues/29

ginkogruen commented:
So I've rebased onto main, reworded my initial commit (which now closes the issue) and changed the relevant parts of the documentation (Where I hope I didn't miss anything).

Is there anything else which needs to be done?
